src/main/java/org/logstash/beats/BatchEncoder.java

```diff
@@ -0,0 +1,69 @@
+package org.logstash.beats;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.MessageToByteEncoder;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.util.Map;
+
+
+public class BatchEncoder extends MessageToByteEncoder<Batch> {
+    private final static Logger logger = LogManager.getLogger(BatchEncoder.class.getName());
+
+
+    @Override
+    protected void encode(ChannelHandlerContext ctx, Batch batch, ByteBuf out) throws Exception {
+        out.writeByte(batch.getProtocol());
+        out.writeByte('W');
+        out.writeInt(batch.size());
+        out.writeBytes(getPayload(ctx, batch));
+    }
+
+    protected ByteBuf getPayload(ChannelHandlerContext ctx, Batch batch) throws IOException {
+        ByteBuf payload = ctx.alloc().buffer();
```

Comment:
Thanks will verify that, I think the name of the plugin you are talking is `Buffer Monitor` I just installed it.